internal/drivers/postgresql/sql.go

```diff
@@ -40,87 +43,115 @@ func quoteBytes(buf []byte) string {
 func quoteValue(arg any) (str string) {
 	switch arg := arg.(type) {
 	case nil:
-		str = "null"
+		return "null"
 	case int:
-		str = strconv.FormatInt(int64(arg), 10)
+		return strconv.FormatInt(int64(arg), 10)
 	case int8:
-		str = strconv.FormatInt(int64(arg), 10)
+		return strconv.FormatInt(int64(arg), 10)
 	case int16:
-		str = strconv.FormatInt(int64(arg), 10)
+		return strconv.FormatInt(int64(arg), 10)
 	case int32:
-		str = strconv.FormatInt(int64(arg), 10)
+		return strconv.FormatInt(int64(arg), 10)
 	case *int32:
 		if arg == nil {
-			str = "null"
+			return "null"
 		} else {
-			str = strconv.FormatInt(int64(*arg), 10)
+			return strconv.FormatInt(int64(*arg), 10)
 		}
 	case int64:
-		str = strconv.FormatInt(arg, 10)
+		return strconv.FormatInt(arg, 10)
 	case *int64:
 		if arg == nil {
-			str = "null"
+			return "null"
 		} else {
-			str = strconv.FormatInt(*arg, 10)
+			return strconv.FormatInt(*arg, 10)
 		}
 	case float32:
-		str = strconv.FormatFloat(float64(arg), 'f', -1, 32)
+		return strconv.FormatFloat(float64(arg), 'f', -1, 32)
 	case float64:
-		str = strconv.FormatFloat(arg, 'f', -1, 64)
+		return strconv.FormatFloat(arg, 'f', -1, 64)
 	case *float64:
 		if arg == nil {
-			str = "null"
+			return "null"
 		} else {
-			str = strconv.FormatFloat(*arg, 'f', -1, 64)
+			return strconv.FormatFloat(*arg, 'f', -1, 64)
 		}
 	case bool:
-		str = strconv.FormatBool(arg)
+		return strconv.FormatBool(arg)
 	case *bool:
 		if arg == nil {
-			str = "null"
+			return "null"
 		} else {
-			str = strconv.FormatBool(*arg)
+			return strconv.FormatBool(*arg)
 		}
 	case []byte:
-		str = quoteBytes(arg)
+
+		return quoteBytes(arg)
 	case *string:
 		if arg == nil {
-			str = "null"
-		} else {
-			str = quoteString(*arg)
+			return "null"
+		}
+		if util.IsJSON(*arg) {
+			return *arg
 		}
+		return quoteString(*arg)
+
 	case string:
-		str = quoteString(arg)
+		if util.IsJSON(arg) {
+			return arg
+		}
+		return quoteString(arg)
 	case *time.Time:
 		if arg == nil {
-			str = "null"
+			return "null"
 		} else {
-			str = (*arg).Truncate(time.Microsecond).Format("'2006-01-02 15:04:05.999999999Z07:00:00'")
+			return (*arg).Truncate(time.Microsecond).Format("'2006-01-02 15:04:05.999999999Z07:00:00'")
 		}
 	case time.Time:
-		str = arg.Truncate(time.Microsecond).Format("'2006-01-02 15:04:05.999999999Z07:00:00'")
+		return arg.Truncate(time.Microsecond).Format("'2006-01-02 15:04:05.999999999Z07:00:00'")
 	case []string:
 		var ns []string
 		for _, thes := range arg {
 			ns = append(ns, pq.QuoteLiteral(thes))
 		}
-		str = "ARRAY[" + strings.Join(ns, ",") + "]"
+		return "ARRAY[" + strings.Join(ns, ",") + "]::JSONB]"
+	case map[string]interface{}:
+		if len(arg) == 0 {
+			return "null"
+		}
+
+		return quoteString(util.JSONStringify(arg))
+
 	case []interface{}:
-		str = quoteString(util.JSONStringify(arg))
+		if len(arg) == 0 {
```

Comment:
can't do this because we need empty `[]` which is valid JSON and trying to import our test shop fails if we add this (i removed it yesterday)